src/plugins/intel_gpu/src/kernel_selector/kernel_base_opencl.cpp

```diff
@@ -173,6 +173,7 @@ std::shared_ptr<KernelString> KernelBaseOpenCL::GetKernelString(const std::strin
         kernel_string->undefs = jit.second;
         if (engine_info.vendor_id == cldnn::INTEL_VENDOR_ID) {
             kernel_string->options = exe_mode + " -cl-mad-enable";
+            kernel_string->options += " -cl-intel-greater-than-4GB-buffer-required";
```

Comment:
This is changing global configuration. I'm afraid it may have perf impact for other workloads.

Reply:
Yes, the question to estimate the penalty, if any

Reply:
The performance penalty was quite noticeable. I think we need to introduce additional config to enable this feature.